lib/datadog/ci/ext/environment.rb

```diff
@@ -447,14 +454,23 @@ def extract_local_git
 
         def branch_or_tag(branch_or_tag)
           branch = tag = nil
-          if branch_or_tag.include?('tags/')
+          if !branch_or_tag.nil? && branch_or_tag.include?('tags/')
             tag = branch_or_tag
           else
             branch = branch_or_tag
           end
 
           [branch, tag]
         end
+
+        def extract_name_email(name_and_email)
+          if name_and_email.include? '<'
+            name_and_email =~ /^([^<]*)<([^>]*)>$/
+            [Regexp.last_match(1).strip, Regexp.last_match(2)]
```

Comment:
```suggestion
            match = /^([^<]*)<([^>]*)>$/.match(name_and_email)
            [match[1].strip, match[2]]
```

Reply:
I think this will also help the Sorbet failure in CI.

Reply:
thanks! It seems to be still failing though

Reply:
The typechecker is pointing out that if `name_and_email` does not match the given regexp, the result of calling `#match` is going to be `nil`, and thus `match[1]` and `match[2]` will cause failures.

To fix this, we can add a nil check to `match` before actually using it OR alternatively if we think sorbet is not being relevant here, we can add `# typed: false` at the top of the file to turn it off.

(Sorbet actually provides a library to give type hints, for instance `T.must(match)` to say "I expect that match is always non-nil here", but we don't want to depend on it at this time, so that's not an option for us).

Reply:
I think the nil check to `match` makes perfect sense. How would you do it?

Reply:
Perhaps something like this:

```ruby
def extract_name_email(name_and_email)
  if name_and_email.include?('<') && (match = /^([^<]*)<([^>]*)>$/.match(name_and_email))
    name = match[1].strip if match[1]
    email = match[2]
    return [name, email] if name && email
  end
  
  [nil, name_and_email]
end
```

If we could write a bit more modern Ruby this could be made cleaner, but unfortunately we still need to support Ruby 2.1 which doesn't have a lot of the niceties that were introduced in the past 8 years.

Reply:
awesome, thanks Ivo. I'll implement it like this

Reply:
that didn't work either, neither did my own update 😆

Reply:
Oops, I guess sorbet doesn't accept/see that the result of `match[1]` doesn't change from call to call, and thus that it's ok to do `match[1].strip if match[1]`.

I've pushed an alternative version onto the branch. Btw a small tip is that you can run sorbet locally (`bundle exec rake typecheck` or `bundle exec srb tc`) to iterate faster without needing to wait for CI.

Reply:
thanks! We did it, it looks like :-)